packages/itwinui-css/src/dialog/dialog.scss

```diff
@@ -144,11 +144,14 @@ $iui-dialog-width: 380px;
   line-height: 1.5;
 }
 
+@mixin iui-dialog-title-close-button {
+  margin-left: auto;
```

Comment:
I would have still gone with margin on title. Is it really a con that you can't have anything on the immediate right of the title? The user could put their stuff inside the title.

Plus the same con would also apply to the close icon, as you can't easily put stuff on the immediate left of the icon.

Reply:
But there are less chances that you would want to put something next to the close icon. And putting inside a title doesn't work normally as you need additional classes to center things inside.

Reply:
> But there are less chances that you would want to put something next to the close icon

I can't find the screenshot but @FlyersPh9 had a request from someone who wanted an icon right next to the close icon.

> And putting inside a title doesn't work normally as you need additional classes to center things inside.

Maybe we should handle that then? It would also help towards https://github.com/iTwin/iTwinUI/issues/586

Reply:
We've had 2 requests that i can recall:
1. An icon in front / to the left of the dialog title text.
2. More than 1 borderless button where the close icon is located.

Reply:
Ok, moved margin to the title and added some centering logic to it.